### PR TITLE
Refactored build infrastructure

### PR DESCRIPTION
**Pull request by Andarist**
Profits? Smaller build (and less bytes added to the host app, by removing uuid dependency), better tree-shakeability for upcoming webpack@4.

### CODE REVIEW

**Comment by wyze** [comment]
Looks lovely! Do you know what the size was before and after this change? Hard for me to tell as the `uuid` dependency was missing, so I couldn't use bundlephobia.com for ease of use.

**Comment by Andarist** [comment]
It was a little bit hard to check, because it didnt build a single umd file. Could try to compare, but dunno if its worth the effort. Removing `uuid` alone is worth the change in regards of the size.

**Comment by wyze** [comment]
Ah okay, no worries!

File: src/uid.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+// @flow
+export default (): string => Math.random().toString(36).substring(2)
```
Comment (AjayPoshak) [inline src/uid.js]: Cool 👍

**Comment by danilowoz** [comment]
Hey @Andarist!
Your changes are so great! Thank you so much!

I just didn't understand one thing, you removed the `.npmignore`, isn't it necessary?

**Comment by wyze** [comment]
@danilowoz It was dropped in favor of using the `files` property in package.json, which says which files to include when you run `npm publish`.

https://github.com/danilowoz/react-content-loader/pull/68/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R26

**Comment by Andarist** [comment]
Exactly, this is ofc just a proposal from my side - but I always find `files` being just more maintainable than `.npmignore`. If you want I can revert this change.

File: src/uid.js (hunk 1 above)
Comment (Andarist) [reply src/uid.js]: stolen from redux 😄  - https://github.com/reactjs/redux/blob/4c9cbd15839d153e4aee9f935c11ec9ca4af10b3/src/utils/actionTypes.js#L10-L14

**Comment by Andarist** [comment]
> I don't see any breaking change so this should ship transparently, do you agree @Andarist?

I believe so, I had maximum compatibility in mind.

**Comment by danilowoz** [comment]
Nice, I agree with everything!
Thank you guys, you are awesome!

**Pull request merged**